tensorflow/core/grappler/optimizers/generic_layout_optimizer_test.cc

```diff
@@ -49,6 +49,31 @@ constexpr int kDepthIn = 8;
 constexpr int kKernel = 3;
 constexpr int kDepthOut = 16;
 
+#if (GOOGLE_CUDA || TENSORFLOW_USE_ROCM)
+#define DIMS(n, h, w, c) \
```

Comment:
Could you please add few lines of documentation explaining what's going on here, so people without a context can get what's up quickly. Something like:

"When GPU available we test .... when only CPU is available we ..."

Reply:
OK. I will follow-up as I am covering the developer who is on vacation now